server/src/test/java/org/apache/druid/segment/indexing/DataSchemaTest.java

```diff
@@ -324,109 +326,34 @@ public void testEmptyDatasource()
     );
   }
 
-  @Test
-  public void testNewlineDatasource()
-  {
-    Map<String, Object> parser = jsonMapper.convertValue(
-        new StringInputRowParser(
-            new JSONParseSpec(
-                new TimestampSpec("time", "auto", null),
-                new DimensionsSpec(
-                    DimensionsSpec.getDefaultSchemas(ImmutableList.of("time", "dimA", "dimB", "col2")),
-                    ImmutableList.of("dimC"),
-                    null
-                ),
-                null,
-                null
-            ),
-            null
-        ), JacksonUtils.TYPE_REFERENCE_MAP_STRING_OBJECT
-    );
-
-    expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage(
-        "dataSource cannot contain whitespace character."
-    );
-
-    DataSchema schema = new DataSchema(
-        "new\nline\ncharacter",
-        parser,
-        null,
-        null,
-        null,
-        jsonMapper
-    );
-  }
-
 
   @Test
-  public void testCarriageReturnDatasource()
+  public void testWhitespaceDatasource()
   {
-    Map<String, Object> parser = jsonMapper.convertValue(
-        new StringInputRowParser(
-            new JSONParseSpec(
-                new TimestampSpec("time", "auto", null),
-                new DimensionsSpec(
-                    DimensionsSpec.getDefaultSchemas(ImmutableList.of("time", "dimA", "dimB", "col2")),
-                    ImmutableList.of("dimC"),
-                    null
-                ),
-                null,
-                null
-            ),
-            null
-        ), JacksonUtils.TYPE_REFERENCE_MAP_STRING_OBJECT
-    );
-
-    expectedException.expect(IllegalArgumentException.class);
-    expectedException.expectMessage(
-        "dataSource cannot contain whitespace character."
-    );
-
-    DataSchema schema = new DataSchema(
-        "carriage\return\rcharacter",
-        parser,
-        null,
-        null,
-        null,
-        jsonMapper
-    );
+    List<String> dataSource = Arrays.asList("\tab\t", "\rcarriage\return\r", "\nnew\nline\n");
+    for (String name : dataSource) {
+      testDatasource(name);
+    }
   }
 
-  @Test
-  public void testTabDatasource()
+  private void testDatasource(String dataSource)
```

Comment:
Consider renaming to be less generic or inlining it above since it's now only called once